spring-xd-dirt/src/main/java/org/springframework/xd/dirt/server/admin/deployment/zk/ZKDeploymentMessagePublisher.java

```diff
@@ -33,21 +83,164 @@
 	 * @param deploymentQueue the deployment queue
 	 */
 	public ZKDeploymentMessagePublisher(DeploymentQueue deploymentQueue) {
-		this.distributedQueue = deploymentQueue.getDistributedQueue();
+		this.deploymentQueue = deploymentQueue;
+	}
+
+	private CuratorFramework getClient() {
+		return this.deploymentQueue.getClient();
 	}
 
 	/**
-	 * Produces the deployment message into ZK distributed queue.
+	 * Return the amount of time in milliseconds to wait for the message
+	 * consumer to process the deployment message. This value is 3 times
+	 * the value of the configured module deployment timeout.
 	 *
-	 * @param deploymentMessage the deployment message
+	 * @see #deploymentTimeout
+	 * @return timeout for deployment message processing
 	 */
+	private long getTimeout() {
+		return deploymentTimeout * 3;
```

Comment:
@pperalta Added some logic that calculates timeout based on the modules in context. There are still a few items to address. Before that, I wanted to make sure this is in the right direction: https://github.com/ilayaperumalg/spring-xd/commit/40b0f43979d7a96daf37fad9dd4fba707759ce78